### DISCUSSION TITLE
upgrading the package may lead to version incompatibilities?

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#222] [Q&A] [answered]
I was using a script that gave me:

> Traceback (most recent call last):
  File "../../code/getr.py", line 5, in <module>
    from ChildProject.metrics import gamma, segments_to_grid, grid_to_vector, vectors_to_annotation_task
ModuleNotFoundError: No module named 'ChildProject.metrics'

So I checked my versions:
datalad 0.14.0
ChildProject-0.0.1-py3.6.egg-inf

I figured perhaps ChildProject was old, so I thought I needed to force upgrade:

```
 pip3 uninstall ChildProject
 pip3 install git+https://github.com/LAAC-LSCP/ChildProject.git
```

> Collecting git+https://github.com/LAAC-LSCP/ChildProject.git
  Cloning https://github.com/LAAC-LSCP/ChildProject.git to /private/var/folders/mx/f_gpzmq105x8sgqfdtbr4zrr0000gn/T/pip-req-build-1fhifnck
Requirement already satisfied: pandas>=0.25.0 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (1.1.4)
Requirement already satisfied: jinja2 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (2.11.2)
Requirement already satisfied: numpy>=1.16.5 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (1.19.4)
Requirement already satisfied: sox in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (1.4.1)
Requirement already satisfied: datalad in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (0.14.0)
Requirement already satisfied: requests<2.25.0 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (2.24.0)
Requirement already satisfied: lxml in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (4.6.2)
Requirement already satisfied: pympi-ling in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (1.69)
Collecting pylangacq (from ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/cd/ed/932dd6a1fb069944ea28e13cbc7f1d8aae09aef82af2a3dc14ddb6fad155/pylangacq-0.14.1-py3-none-any.whl (68kB)
    100% |████████████████████████████████| 71kB 2.0MB/s 
Requirement already satisfied: python-dateutil>=2.8.1 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (2.8.1)
Collecting librosa (from ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/54/19/a0e2bdc94bc0d1555e4f9bc4099a0751da83fa6e1e6157ec005564f8a98a/librosa-0.8.1-py3-none-any.whl (203kB)
    100% |████████████████████████████████| 204kB 2.0MB/s 
Requirement already satisfied: pydub in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (0.24.1)
Collecting pysoundfile (from ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/f7/62/bddd31ba4f575eca81c6642d809fe01ed5ee5f55ebb4e9b14d0a033cc6bf/PySoundFile-0.9.0.post1-py2.py3.cp26.cp27.cp32.cp33.cp34.cp35.cp36.pp27.pp32.pp33-none-macosx_10_5_x86_64.macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.whl (573kB)
    100% |████████████████████████████████| 573kB 2.1MB/s 
Collecting nltk (from ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/5e/37/9532ddd4b1bbb619333d5708aaad9bf1742f051a664c3c6fa6632a105fd8/nltk-3.6.2-py3-none-any.whl (1.5MB)
    100% |████████████████████████████████| 1.5MB 22.9MB/s 
Collecting sklearn (from ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/1e/7a/dbb3be0ce9bd5c8b7e3d87328e79063f8b263b2b1bfa4774cb1147bfcd3f/sklearn-0.0.tar.gz
Requirement already satisfied: PyYAML in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (5.4.1)
Requirement already satisfied: panoptes-client in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from ChildProject==0.0.1) (1.3.0)
Collecting importlib-resources (from ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/a4/30/b230b6586bcf6b80752ae42979f3b0da70bbde977d2b73eafd20c693b3db/importlib_resources-5.1.4-py3-none-any.whl
Collecting pygamma-agreement (from ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/cc/32/95f25c97827a4fc0de9d981d7082296767363ca0d09bd07725a96a13191d/pygamma_agreement-0.1.6-py3-none-any.whl
Requirement already satisfied: pytz>=2017.2 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from pandas>=0.25.0->ChildProject==0.0.1) (2020.4)
Requirement already satisfied: MarkupSafe>=0.23 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from jinja2->ChildProject==0.0.1) (1.1.1)
Requirement already satisfied: appdirs in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (1.4.4)
Requirement already satisfied: chardet>=3.0.4 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (3.0.4)
Requirement already satisfied: iso8601 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (0.1.13)
Requirement already satisfied: humanize in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (3.1.0)
Requirement already satisfied: fasteners>=0.14 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (0.15)
Requirement already satisfied: patool>=1.7 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (1.12)
Requirement already satisfied: tqdm in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (4.52.0)
Requirement already satisfied: wrapt in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (1.12.1)
Requirement already satisfied: annexremote in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (1.4.3)
Requirement already satisfied: boto in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (2.49.0)
Requirement already satisfied: keyring>=8.0 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (21.5.0)
Requirement already satisfied: keyrings.alt in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (4.0.1)
Requirement already satisfied: msgpack in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (1.0.0)
Requirement already satisfied: jsmin in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (2.2.2)
Requirement already satisfied: PyGithub in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (1.53)
Requirement already satisfied: simplejson in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (3.17.2)
Requirement already satisfied: whoosh in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from datalad->ChildProject==0.0.1) (2.7.4)
Requirement already satisfied: certifi>=2017.4.17 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from requests<2.25.0->ChildProject==0.0.1) (2020.11.8)
Requirement already satisfied: idna<3,>=2.5 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from requests<2.25.0->ChildProject==0.0.1) (2.10)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from requests<2.25.0->ChildProject==0.0.1) (1.25.11)
Collecting dataclasses; python_version < "3.7" (from pylangacq->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/fe/ca/75fac5856ab5cfa51bbbcefa250182e50441074fdc3f803f6e76451fab43/dataclasses-0.8-py3-none-any.whl
Requirement already satisfied: six>=1.5 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from python-dateutil>=2.8.1->ChildProject==0.0.1) (1.15.0)
Collecting joblib>=0.14 (from librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/55/85/70c6602b078bd9e6f3da4f467047e906525c355a4dacd4f71b97a35d9897/joblib-1.0.1-py3-none-any.whl (303kB)
    100% |████████████████████████████████| 307kB 2.2MB/s 
Collecting soundfile>=0.10.2 (from librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/13/63/073fc76f2405af3ad1a2e667ac237a527fcdd81d4cbc7e2b721727687453/SoundFile-0.10.3.post1-py2.py3.cp26.cp27.cp32.cp33.cp34.cp35.cp36.pp27.pp32.pp33-none-macosx_10_5_x86_64.macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.whl (613kB)
    100% |████████████████████████████████| 614kB 2.4MB/s 
Collecting packaging>=20.0 (from librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/3e/89/7ea760b4daa42653ece2380531c90f64788d979110a2ab51049d92f408af/packaging-20.9-py2.py3-none-any.whl (40kB)
    100% |████████████████████████████████| 40kB 1.9MB/s 
Collecting audioread>=2.0.0 (from librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/b3/d1/e324634c5867a668774d6fe233a83228da4ba16521e19059c15df899737d/audioread-2.1.9.tar.gz (377kB)
    100% |████████████████████████████████| 378kB 2.4MB/s 
Collecting pooch>=1.0 (from librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/40/b9/9876662636ba451d4406543047c0b45ca5b4e830f931308c8274dad1db43/pooch-1.3.0-py3-none-any.whl (51kB)
    100% |████████████████████████████████| 61kB 2.2MB/s 
Collecting numba>=0.43.0 (from librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/af/a4/de3f7b6610f7a840e12ba1723ce9d3c10c41f070b86f66016b67249add78/numba-0.53.1-cp36-cp36m-macosx_10_14_x86_64.whl (2.2MB)
    100% |████████████████████████████████| 2.2MB 2.4MB/s 
Collecting decorator>=3.0.0 (from librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/6a/36/b1b9bfdf28690ae01d9ca0aa5b0d07cb4448ac65fb91dc7e2d094e3d992f/decorator-5.0.9-py3-none-any.whl
Collecting resampy>=0.2.2 (from librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/79/75/e22272b9c2185fc8f3af6ce37229708b45e8b855fd4bc38b4d6b040fff65/resampy-0.2.2.tar.gz (323kB)
    100% |████████████████████████████████| 327kB 2.2MB/s 
Collecting scipy>=1.0.0 (from librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/47/c7/348acee81b0cf8eec66b4a71c8cca188f405061cb76cc3f9f72249568a22/scipy-1.5.4-cp36-cp36m-macosx_10_9_x86_64.whl (28.8MB)
    100% |████████████████████████████████| 28.8MB 1.1MB/s 
Collecting scikit-learn!=0.19.0,>=0.14.0 (from librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/68/c6/5ed3f5b4eb2b0f99ec2e8f9bd2030020083323a42da5afa9f9a29e50ede6/scikit_learn-0.24.2-cp36-cp36m-macosx_10_13_x86_64.whl (7.2MB)
    100% |████████████████████████████████| 7.2MB 1.9MB/s 
Collecting cffi>=0.6 (from pysoundfile->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/4d/3d/7cbf1aa7f34de5ecf18b1b63cbc3250911e85567f0eb4f97d7f03ebc165b/cffi-1.14.5-cp36-cp36m-macosx_10_9_x86_64.whl (176kB)
    100% |████████████████████████████████| 184kB 2.3MB/s 
Collecting regex (from nltk->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/a0/bf/75584fbc47a32a778e25ab74f0e5db9f8c3c8ac6978d25f952fd9ee134f0/regex-2021.4.4-cp36-cp36m-macosx_10_9_x86_64.whl (285kB)
    100% |████████████████████████████████| 286kB 2.4MB/s 
Collecting click (from nltk->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/76/0a/b6c5f311e32aeb3b406e03c079ade51e905ea630fc19d1262a46249c1c86/click-8.0.1-py3-none-any.whl (97kB)
    100% |████████████████████████████████| 102kB 2.3MB/s 
Requirement already satisfied: future<0.19,>=0.16 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from panoptes-client->ChildProject==0.0.1) (0.18.2)
Requirement already satisfied: python-magic<0.5,>=0.4 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from panoptes-client->ChildProject==0.0.1) (0.4.20)
Requirement already satisfied: redo>=1.7 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from panoptes-client->ChildProject==0.0.1) (2.0.4)
Requirement already satisfied: zipp>=3.1.0; python_version < "3.10" in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from importlib-resources->ChildProject==0.0.1) (3.4.0)
Collecting cvxpy==1.0.25 (from pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/09/a4/bd2585aa6024393fe4ed9346a9b66500e8e0e054d0ff4356c094fb7fbf41/cvxpy-1.0.25-cp36-cp36m-macosx_10_7_x86_64.whl (676kB)
    100% |████████████████████████████████| 686kB 1.3MB/s 
Requirement already satisfied: typing-extensions>=3.7.4.3 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from pygamma-agreement->ChildProject==0.0.1) (3.10.0.0)
Collecting pyannote.database>=4.0.1 (from pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/f8/cb/82d8035dbc3dc9aac80b5475fcac63125fc3669e4d8d07261ee3d8507f57/pyannote.database-4.1-py3-none-any.whl (41kB)
    100% |████████████████████████████████| 51kB 2.9MB/s 
Collecting pyannote.core>=4.1 (from pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/ba/09/64e64cedccf764be513405d1ff3d3d2bbcbea60d18fb6a6d9e815aedcd79/pyannote.core-4.1-py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 2.3MB/s 
Collecting TextGrid>=1.5 (from pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/9f/9e/04fb27ec5ac287b203afd5b228bc7c4ec5b7d3d81c4422d57847e755b0cc/TextGrid-1.5-py3-none-any.whl
Collecting sortedcontainers>=2.0.4 (from pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/32/46/9cb0e58b2deb7f82b84065f37f3bffeb12413f947f9388e4cac22c4621ce/sortedcontainers-2.4.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from humanize->datalad->ChildProject==0.0.1) (39.0.1)
Requirement already satisfied: monotonic>=0.1 in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from fasteners>=0.14->datalad->ChildProject==0.0.1) (1.5)
Requirement already satisfied: importlib-metadata>=1; python_version < "3.8" in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from keyring>=8.0->datalad->ChildProject==0.0.1) (2.0.0)
Requirement already satisfied: pyjwt in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from PyGithub->datalad->ChildProject==0.0.1) (1.7.1)
Requirement already satisfied: deprecated in /Users/acristia/ChildProjectVenv/lib/python3.6/site-packages (from PyGithub->datalad->ChildProject==0.0.1) (1.2.10)
Collecting pyparsing>=2.0.2 (from packaging>=20.0->librosa->ChildProject==0.0.1)
  Using cached https://files.pythonhosted.org/packages/8a/bb/488841f56197b13700afd5658fc279a2025a39e22449b7cf29864669b15d/pyparsing-2.4.7-py2.py3-none-any.whl
Collecting llvmlite<0.37,>=0.36.0rc1 (from numba>=0.43.0->librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/f0/74/f509fe2ada2875db48e09a4bbcc8a304a6d908adab443308768d1963de7c/llvmlite-0.36.0-cp36-cp36m-macosx_10_9_x86_64.whl (18.5MB)
    100% |████████████████████████████████| 18.5MB 1.4MB/s 
Collecting threadpoolctl>=2.0.0 (from scikit-learn!=0.19.0,>=0.14.0->librosa->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/f7/12/ec3f2e203afa394a149911729357aa48affc59c20e2c1c8297a60f33f133/threadpoolctl-2.1.0-py3-none-any.whl
Collecting pycparser (from cffi>=0.6->pysoundfile->ChildProject==0.0.1)
  Using cached https://files.pythonhosted.org/packages/ae/e7/d9c3a176ca4b02024debf82342dab36efadfc5776f9c8db077e8f6e71821/pycparser-2.20-py2.py3-none-any.whl
Collecting scs>=1.1.3 (from cvxpy==1.0.25->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/12/bd/1ab6a3b3f2791741e6e7c142f932ea1808277e92167e322dec43271b2225/scs-2.1.3.tar.gz (147kB)
    100% |████████████████████████████████| 153kB 190kB/s 
Collecting osqp>=0.4.1 (from cvxpy==1.0.25->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/47/f0/9546e8771addba6b6c22d76f33567293ae1de3f3900d552e4692f60a0549/osqp-0.6.2.post0-cp36-cp36m-macosx_10_9_x86_64.whl (163kB)
    100% |████████████████████████████████| 174kB 2.4MB/s 
Collecting multiprocess (from cvxpy==1.0.25->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/8f/dc/426a82723c460cfab653ebb717590103d6e38cebc9d1f599b0898915ac1d/multiprocess-0.70.11.1-py36-none-any.whl (101kB)
    100% |████████████████████████████████| 102kB 1.9MB/s 
Collecting ecos>=2 (from cvxpy==1.0.25->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/2c/6b/9ef18307078597fb3c8a16be5bcc97180d2eac5032057a182b06e50a15d2/ecos-2.0.7.post1-cp36-cp36m-macosx_10_13_x86_64.whl (71kB)
    100% |████████████████████████████████| 81kB 2.6MB/s 
Collecting typer[all]>=0.2.1 (from pyannote.database>=4.0.1->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/90/34/d138832f6945432c638f32137e6c79a3b682f06a63c488dcfaca6b166c64/typer-0.3.2-py3-none-any.whl
Collecting matplotlib>=2.0.0 (from pyannote.core>=4.1->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/77/bd/a70d352f3c24bd444f3cf182f232cca421de23098fba2cf34df1defa68e6/matplotlib-3.3.4-cp36-cp36m-macosx_10_9_x86_64.whl (8.5MB)
    100% |████████████████████████████████| 8.5MB 1.7MB/s 
Collecting qdldl (from osqp>=0.4.1->cvxpy==1.0.25->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/e5/e3/f1740c49254e674fb8c5bc41bd3051490616057bd2bd466d0f1f93cbf45e/qdldl-0.1.5.post0-cp36-cp36m-macosx_10_9_x86_64.whl (86kB)
    100% |████████████████████████████████| 92kB 2.2MB/s 
Collecting dill>=0.3.3 (from multiprocess->cvxpy==1.0.25->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/52/d6/79f40d230895fa1ce3b6af0d22e0ac79c65175dc069c194b79cc8e05a033/dill-0.3.3-py2.py3-none-any.whl (81kB)
    100% |████████████████████████████████| 81kB 2.1MB/s 
Collecting colorama<0.5.0,>=0.4.3; extra == "all" (from typer[all]>=0.2.1->pyannote.database>=4.0.1->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/44/98/5b86278fbbf250d239ae0ecb724f8572af1c91f4a11edf4d36a206189440/colorama-0.4.4-py2.py3-none-any.whl
Collecting shellingham<2.0.0,>=1.3.0; extra == "all" (from typer[all]>=0.2.1->pyannote.database>=4.0.1->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/76/94/7a764d57d0f46534e0022e651da6547bc5cfe7b6372e7e0ed1dde6f5cb67/shellingham-1.4.0-py2.py3-none-any.whl
Collecting kiwisolver>=1.0.1 (from matplotlib>=2.0.0->pyannote.core>=4.1->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/94/4e/f90bcdd273194230f9c08ec63a6d0874ef36ff5c077f487f761221361cc4/kiwisolver-1.3.1-cp36-cp36m-macosx_10_9_x86_64.whl (61kB)
    100% |████████████████████████████████| 61kB 2.2MB/s 
Collecting cycler>=0.10 (from matplotlib>=2.0.0->pyannote.core>=4.1->pygamma-agreement->ChildProject==0.0.1)
  Using cached https://files.pythonhosted.org/packages/f7/d2/e07d3ebb2bd7af696440ce7e754c59dd546ffe1bbe732c8ab68b9c834e61/cycler-0.10.0-py2.py3-none-any.whl
Collecting pillow>=6.2.0 (from matplotlib>=2.0.0->pyannote.core>=4.1->pygamma-agreement->ChildProject==0.0.1)
  Downloading https://files.pythonhosted.org/packages/14/e9/9c91f2f5d6102eae2051b28f85f3eaad4bdd9c67ede8c4d71170960391fa/Pillow-8.2.0-cp36-cp36m-macosx_10_10_x86_64.whl (2.8MB)
    100% |████████████████████████████████| 2.8MB 2.3MB/s 
typer 0.3.2 has requirement click<7.2.0,>=7.1.1, but you'll have click 8.0.1 which is incompatible.
Installing collected packages: dataclasses, pylangacq, joblib, pycparser, cffi, soundfile, pyparsing, packaging, audioread, pooch, llvmlite, numba, decorator, scipy, resampy, threadpoolctl, scikit-learn, librosa, pysoundfile, regex, click, nltk, sklearn, importlib-resources, scs, qdldl, osqp, dill, multiprocess, ecos, cvxpy, colorama, shellingham, typer, kiwisolver, cycler, pillow, matplotlib, sortedcontainers, pyannote.core, pyannote.database, TextGrid, pygamma-agreement, ChildProject
  Running setup.py install for audioread ... done
  Running setup.py install for resampy ... done
  Running setup.py install for sklearn ... done
  Running setup.py install for scs ... done
  Running setup.py install for ChildProject ... done
Successfully installed ChildProject-0.0.1 TextGrid-1.5 audioread-2.1.9 cffi-1.14.5 click-8.0.1 colorama-0.4.4 cvxpy-1.0.25 cycler-0.10.0 dataclasses-0.8 decorator-5.0.9 dill-0.3.3 ecos-2.0.7.post1 importlib-resources-5.1.4 joblib-1.0.1 kiwisolver-1.3.1 librosa-0.8.1 llvmlite-0.36.0 matplotlib-3.3.4 multiprocess-0.70.11.1 nltk-3.6.2 numba-0.53.1 osqp-0.6.2.post0 packaging-20.9 pillow-8.2.0 pooch-1.3.0 pyannote.core-4.1 pyannote.database-4.1 pycparser-2.20 pygamma-agreement-0.1.6 pylangacq-0.14.1 pyparsing-2.4.7 pysoundfile-0.9.0.post1 qdldl-0.1.5.post0 regex-2021.4.4 resampy-0.2.2 scikit-learn-0.24.2 scipy-1.5.4 scs-2.1.3 shellingham-1.4.0 sklearn-0.0 sortedcontainers-2.4.0 soundfile-0.10.3.post1 threadpoolctl-2.1.0 typer-0.3.2

-- which fixed my original error (now I have childproject.metrics), but I now get an error when I try to run the sampler:

> Traceback (most recent call last):
  File "/Users/acristia/ChildProjectVenv/bin/child-project", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3088, in <module>
    @_call_aside
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3072, in _call_aside
    f(*args, **kwargs)
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3101, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 576, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 589, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 783, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (click 8.0.1 (/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages), Requirement.parse('click<7.2.0,>=7.1.1'), {'typer'})

I think this may not really be a question for here, but if there are any tips to try to keep all the mutually compatible versions together, that would be great!

**Comment by lucasgautheron** [comment, marked as answer]
Can you try:

```bash
pip3 install git+https://github.com/LAAC-LSCP/ChildProject.git --upgrade
```

?

**Reply by alecristia** [reply]
that ran ok, no errors other than the same as before: "typer 0.3.2 has requirement click<7.2.0,>=7.1.1, but you'll have click 8.0.1 which is incompatible."

and so the sampler throws the same error as before:

Traceback (most recent call last):
  File "/Users/acristia/ChildProjectVenv/bin/child-project", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3088, in <module>
    @_call_aside
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3072, in _call_aside
    f(*args, **kwargs)
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3101, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 576, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 589, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages/pkg_resources/__init__.py", line 783, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (click 8.0.1 (/Users/acristia/ChildProjectVenv/lib/python3.6/site-packages), Requirement.parse('click<7.2.0,>=7.1.1'), {'typer'})

**Reply by lucasgautheron** [reply]
One way of solving the problem for you (temporarily) is to do: `pip install "click<7.2.0,>=7.1.1" --upgrade`

I could enforce this constraint into the package setup configuration, BUT I am not sure this is the best way to handle it in the long term...

**Reply by alecristia** [reply]
this is fixed with `pip install "click<7.2.0,>=7.1.1" --upgrade`
